frontend/util.lua

```diff
@@ -400,4 +400,102 @@ function util.splitToArray(str, splitter, capture_empty_entity)
     return result
 end
 
+--- Convert a Unicode codepoint (number) to UTF8 char
+--
+--- @int c Unicode codepoint
+--- @treturn string UTF8 char
+function util.unicodeCodepointToUtf8(c)
+    if c < 128 then
+        return string.char(c)
+    elseif c < 2048 then
+        return string.char(192 + c/64, 128 + c%64)
+    elseif c < 55296 or 57343 < c and c < 65536 then
+        return string.char(224 + c/4096, 128 + c/64%64, 128 + c%64)
+    elseif c < 1114112 then
+        return string.char(240 + c/262144, 128 + c/4096%64, 128 + c/64%64, 128 + c%64)
+    else
+        return util.unicodeCodepointToUtf8(65533) -- U+FFFD REPLACEMENT CHARACTER
+    end
+end
+
+local HTML_ENTITIES_TO_UTF8 = {
+    ["&lt;"] = "<",
+    ["&gt;"] = ">",
+    ["&amp;"] = "&",
+    ["&quot;"] = '"',
+    ["&apos;"] = "'",
+    ["&nbsp;"] = "\xC2\xA0",
+    ["&#(%d+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x) ) end,
+    ["&#x(%x+);"] = function(x) return util.unicodeCodepointToUtf8( tonumber(x,16) ) end,
+}
+--- Replace HTML entities with their UTF8 equivalent in text
+--
+-- Supports only basic ones and those with numbers (no support
+-- for named entities like &eacute;)
+--- @int string text with HTML entities
+--- @treturn string UTF8 text
+function util.htmlEntitiesToUtf8(text)
+    for k,v in pairs(HTML_ENTITIES_TO_UTF8) do
+        text = text:gsub(k, v)
+    end
+    return text
+end
+
+--- Convert HTML to plain text
+--
+--- @string text HTML text
+--- @treturn string plain text
+function util.htmlToPlainText(text)
+    -- Replace <br> and <p> with \n
+    text = text:gsub("%s*<%s*br%s*/?>%s*", "\n") -- <br> and <br/>
+    text = text:gsub("%s*<%s*p%s*>%s*", "\n") -- <p>
+    text = text:gsub("%s*</%s*p%s*>%s*", "\n") -- </p>
+    text = text:gsub("%s*<%s*p%s*/>%s*", "\n") -- standalone <p/>
+    -- Remove all HTML tags
+    text = text:gsub("<[^>]*>", "")
+    -- Convert HTML entities
+    text = util.htmlEntitiesToUtf8(text)
+    -- Trim spaces and new lines at start and end
+    text = text:gsub("^[\n%s]*", "")
+    text = text:gsub("[\n%s]*$", "")
+    return text
+end
+
+--- Remove HTML tags from text if text seems to be HTML
+--
+--- @string text the string with possibly some HTML
+--- @treturn string cleaned text
+function util.htmlToPlainTextIfHtml(text)
+    local is_html = false
+    -- Quick way to check if text is some HTML:
+    -- if it is HTML, we should get the same number of '<' and '>'
+    -- that surround tags
+    local _, nb_gt, nb_lt
+    _, nb_gt = text:gsub(">", "")
```

Comment:
I think counting < and > will introduce more true negative instead of concerning false positive: Html contents contain js will be treated as non-html.
But your example is not accurate, contents contain js are usually Html.